pkgs/development/python-modules/torch/tests.nix

```diff
@@ -28,4 +28,7 @@ rec {
     feature = "rocm";
     libraries = ps: [ ps.torchWithRocm ];
   };
+  tester-darwin = callPackage ./mk-darwin-check.nix {
+    libraries = ps: [ ps.torch ];
+  };
```

Comment:
This will not run the test as part of the derivation though:)

Good idea to move the script into a separate file

It's still a bit disappointing that the script is essentially the same as for cuda/rocm, the difference is just one line which we could simply parameterize (e.g. instead of the `@` decorator syntax, we could `f = compile(f)` under an `if` guard)

Reply:
Can we somehow also add this as part of the tests for the package?

Reply:
Is this good to go in now?